From 5b1a80581e5d95618804a84076d631443bde196e Mon Sep 17 00:00:00 2001
From: Bingwu Zhang <xtex@aosc.io>
Date: Sat, 22 Nov 2025 22:12:59 +0800
Subject: [PATCH 121/133] Compilation: fix incorrect saved buf count
X-Developer-Signature: v=1; a=openpgp-sha256; l=733; i=xtex@astrafall.org;
 h=from:subject; bh=tFW7SDC7CiNr0GA38EkuZVMHaakEfIlaLnzWhUbKWx4=;
 b=owGbwMvMwCW2U4Ij7wZL9ETG02pJDJn78vhylneuXXqgI9b/2fs/M+yfrje1FZ5ymGVVb8Zae
 Yb+PR3nOiayMIhxMViKKbIUGTZ4s+qk84suK5eFmcPKBDJEWqSBgYGBgYWBLzcxr9RIx0jPVNtQ
 z9BQx0jHiIGLUwCmOlyYkWHH2UlpKuefPNeap3JDUPnPgyUOn6cvt9pfVvZ4SuYWrvwWhn/an++
 tdXf3Kks9y9ZiLbjtTHLxV8tts57aRvZ5xqzQ2sQJAA==
X-Developer-Key: i=xtex@astrafall.org; a=openpgp;
 fpr=7231804B052C670F15A6771DB918086ED8045B91

Change-Id: Ib2a2ffcb6224e571fed1151e687d8a8efe0bb95b
(cherry picked from commit fc517bd01c8c60290e903e58f1f567b065ae72a0)
---
 src/Compilation.zig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Compilation.zig b/src/Compilation.zig
index 3752e93a0db0..3523211ac8f4 100644
--- a/src/Compilation.zig
+++ b/src/Compilation.zig
@@ -3722,7 +3722,7 @@ pub fn saveState(comp: *Compilation) !void {
             },
         });
 
-        try bufs.ensureTotalCapacityPrecise(14 + 8 * pt_headers.items.len);
+        try bufs.ensureTotalCapacityPrecise(22 + 9 * pt_headers.items.len);
         addBuf(&bufs, mem.asBytes(&header));
         addBuf(&bufs, @ptrCast(pt_headers.items));
 
-- 
2.52.0

